index.js

```diff
@@ -59,20 +59,26 @@ function ShopifyToken(options) {
   this.apiKey = options.apiKey;
 }
 
+ShopifyToken.prototype.generateNonce = function() {
+  return crypto.randomBytes(16).toString('hex');
+};
+
 /**
  * Build the authorization URL.
  *
  * @param {String} shop The shop name
  * @param {Array|String} [scopes] The list of scopes
+ * @param {String} Random string (ideally persisted elsewhere) for comparison on callback
  * @return {String} The authorization URL
  * @public
  */
-ShopifyToken.prototype.generateAuthUrl = function generateAuthUrl(shop, scopes) {
-  scopes || (scopes = this.scopes);
+ShopifyToken.prototype.generateAuthUrl = function generateAuthUrl(shop, scopes, nonce) {
+  scopes = scopes || this.scopes;
```

Comment:
Allow both `scopes` and `nonce` to be optional. Right now if `scopes` is not provided but `nonce` is, `scopes` will get the `nonce` value.